### ISSUE TITLE
[Improvement]:  move camera to high position when detecting crashes and kicks

### ISSUE THREAD

**Issue by lonelybud** [Yimura/YimMenu#2022]
### Which feature

I was a host in a session.
Somebody started to ragdoll, camera twist, crash, gta banner, kick me etc all at once
I lagged so bad and crashed.

I believe maybe swathing to free cam mode  I could have been able to click the host kick button on that's guy inf page.

### Reason

Moving camera up at least give us time to observe and take decision

### Additional context

Using this build https://github.com/YimMenu/YimMenu/commit/ee364cd684e85b6287b4034fed5d189fc29dbe94

**Comment by Yimura** [comment]
You need to always consider the weight of the offence against the reaction you'll put up against it.

GTA banner is a very light offence an can also be caused by a legit player, so I would only use kick reactions on offences like them trying to kick, crash,... you.

In regards to the crash, any clue on what happened or anything you saw? Any clue on the specific crash? We'd rather fix the crash at the root than to use some makeshift fix.

**Comment by lonelybud** [comment]
my game seems to got stuck as  frame rates were very low (but i held for about 1 min in the lobby and then got desync kicked into a public lobby of my own). After that my game crashed. 
Restarted the game and then i went back into same lobby by spoofing host token and kicked that player. 

This was the guy who kicked me - `Anupam1111`.
I am using `relwithdebinfo` local build at https://github.com/YimMenu/YimMenu/commit/ee364cd684e85b6287b4034fed5d189fc29dbe94

This was the log (123MB) when i was kicked out of that lobby when i was host there -
[08-24-2023-23-16-31_cout.zip](https://github.com/YimMenu/YimMenu/files/12601301/08-24-2023-23-16-31_cout.zip)

no need to signup dropbox. just cancel the popup and click download.

**Comment by xiaoxiao921** [comment]
When you upload big logs, always zip them, their usually see a 95% size decrease

**Comment by 0xEmiil** [comment]
Is this the cam you mean?
![image](https://github.com/YimMenu/YimMenu/assets/79391187/a8b5f825-d4ad-45ca-b047-b0285b64ecbf)

If it is i dont see any reason to use it on protection, the only way to use it is on crashes when you spawn a slod ped, maybe other thing lagged you like PTFX some kind of pool crash you but afaik that can be patched  without using the Sky Cam

**Comment by lonelybud** [comment]
> Is this the cam you mean?

I meant the cam set to high altitude at random location in sky. I did not understood the picture you attached.

**Comment by 0xEmiil** [comment]
> > Is this the cam you mean?
> 
> I meant the cam set to high altitude at random location in sky. I did not understood the picture you attached.

The pic that i attached is lit what you are saying a cam in the highest GTA altitude, idk if i can show full screen cuz its a from another mod that is not yim

**Comment by lonelybud** [comment]
in these files we can make changes.

`struct reactions` is present here for just info.
src\core\settings.hpp

`class reaction` we can add a variable here -> `cameraHigh` something
src\backend\reactions\reaction.hpp

`draw_reaction function` -> here show checkbox for **set camera high** 
src\views\settings\view_reaction_settings.cpp

`process_common function` here set camera to high if checked
src\backend\reactions\reaction.cpp

basically this option should get checked `Freecam` and height of camera automatically set to some random coordinate high.
![image](https://github.com/YimMenu/YimMenu/assets/139460769/e99bce04-1dc4-451a-9a61-1693cc62c2fc)

Its a simple idea though.

**Comment by lonelybud** [comment]
> in these files we can make changes.

i will raise a pr for this.

**Comment by lonelybud** [comment]
how can i run this command in code 
https://github.com/YimMenu/YimMenu/blob/master/src/backend/looped/self/free_cam.cpp

like
```c++
command->refresh();
```

I can see that it is run using this checkbox -
https://github.com/YimMenu/YimMenu/blob/100de3fee55bdd067f5d2022a933709d8a210442/src/views/self/view_self.cpp#L36

which is  basically this function -
https://github.com/YimMenu/YimMenu/blob/100de3fee55bdd067f5d2022a933709d8a210442/src/gui/components/components.hpp#L70

**Comment by lonelybud** [comment]
i checked my logs. It does not seem to be that the modder i mentioned tried to kick me (it seems more likely something happened from outside the session). Its true that I only received `GTA BANNER` from him. 

I dont know why did i mentioned camera twist, ragdoll etc (may be they seemed to happen visually but i cant see them in logs).

The latest master branch commit present in branch at the time was -

ee364cd Ped animations interface #1668 

The repeatable pattern from logs are as follows - 

```

Dumping stacktrace:
[0] menu\src\hooks\player_management\assign_physical_index.cpp L: 98 `big::hooks::assign_physical_index'::`16'::<lambda_1>::operator()
[1] menu\src\fiber_pool.cpp L: 43 big::fiber_pool::fiber_tick
[2] menu\src\fiber_pool.cpp L: 52 big::fiber_pool::fiber_func
[3] menu\src\script.cpp L: 101 big::script::fiber_func
[4] menu\src\script.cpp L: 27 `big::script::script'::`2'::<lambda_1>::operator()
[5] menu\src\script.cpp L: 28 `big::script::script'::`2'::<lambda_1>::<lambda_invoker_cdecl>
[6]	KERNELBASE.dll ChangeTimerQueueTimer
[7]	ntdll.dll RtlUserFiberStart



Dumping stacktrace:
[0] menu\src\hooks\player_management\assign_physical_index.cpp L: 100 `big::hooks::assign_physical_index'::`16'::<lambda_1>::operator()
[1] menu\src\fiber_pool.cpp L: 43 big::fiber_pool::fiber_tick
[2] menu\src\fiber_pool.cpp L: 52 big::fiber_pool::fiber_func
[3] menu\src\script.cpp L: 101 big::script::fiber_func
[4] menu\src\script.cpp L: 27 `big::script::script'::`2'::<lambda_1>::operator()
[5] menu\src\script.cpp L: 28 `big::script::script'::`2'::<lambda_1>::<lambda_invoker_cdecl>
[6]	KERNELBASE.dll ChangeTimerQueueTimer
[7]	ntdll.dll RtlUserFiberStart


Dumping stacktrace:
[0] menu\src\hooks\player_management\assign_physical_index.cpp L: 145 `big::hooks::assign_physical_index'::`16'::<lambda_1>::operator()
[1] menu\src\fiber_pool.cpp L: 43 big::fiber_pool::fiber_tick
[2] menu\src\fiber_pool.cpp L: 52 big::fiber_pool::fiber_func
[3] menu\src\script.cpp L: 101 big::script::fiber_func
[4] menu\src\script.cpp L: 27 `big::script::script'::`2'::<lambda_1>::operator()
[5] menu\src\script.cpp L: 28 `big::script::script'::`2'::<lambda_1>::<lambda_invoker_cdecl>
[6]	KERNELBASE.dll ChangeTimerQueueTimer
[7]	ntdll.dll RtlUserFiberStart



Dumping stacktrace:
[0] menu\src\services\players\player_service.cpp L: 57 big::player_service::get_by_host_token
[1] menu\src\hooks\protections\receive_net_message.cpp L: 92 big::hooks::receive_net_message
[2]	GTA5.exe+0x13E9BCD 0x7FF6C2CB9BCD
[3]	GTA5.exe+0x13E98E6 0x7FF6C2CB98E6
[4]	GTA5.exe+0x14144FB 0x7FF6C2CE44FB
[5]	GTA5.exe+0x10BA554 0x7FF6C298A554
[6]	GTA5.exe+0x161A8E9 0x7FF6C2EEA8E9
[7]	GTA5.exe+0x161A11B 0x7FF6C2EEA11B
[8]	GTA5.exe+0x277F6 0x7FF6C18F77F6
[9]	GTA5.exe+0x20E9B 0x7FF6C18F0E9B
[10]	GTA5.exe+0x1619D24 0x7FF6C2EE9D24
[11]	GTA5.exe+0x1494 0x7FF6C18D1494
[12]	GTA5.exe+0x12F22C3 0x7FF6C2BC22C3
[13]	GTA5.exe+0x180CE38 0x7FF6C30DCE38
[14]	KERNEL32.DLL BaseThreadInitThunk
[15]	ntdll.dll RtlUserThreadStart



Dumping stacktrace:
[0]	GTA5.exe+0x39386BE 0x7FF6C52086BE
[1]	GTA5.exe+0x110A95F 0x7FF6C29DA95F
[2] menu\src\views\players\view_players.cpp L: 26 big::player_button
[3] menu\src\views\players\view_players.cpp L: 134 big::view::players
[4] menu\src\views\core\view_root.cpp L: 11 big::view::root
[5] menu\src\gui.cpp L: 140 big::gui::dx_on_tick
[6] menu\src\renderer.cpp L: 142 big::renderer::on_present
[7] menu\src\hooks\gui\swap_chain_present.cpp L: 14 big::hooks::swapchain_present
[8]	GTA5.exe+0x138334A 0x7FF6C2C5334A
[9]	GTA5.exe+0x7E8549 0x7FF6C20B8549



[0] menu\src\services\players\player_service.cpp L: 57 big::player_service::get_by_host_token
[1] menu\src\hooks\info\get_network_event_data.cpp L: 88 big::hooks::get_network_event_data
[2]	GTA5.exe+0x1147BD2 0x7FF6C2A17BD2
[3]	GTA5.exe+0x113B05E 0x7FF6C2A0B05E
[4]	GTA5.exe+0x16AFC21 0x7FF6C2F7FC21
[5]	GTA5.exe+0x10DA101 0x7FF6C29AA101
[6]	GTA5.exe+0x11148A9 0x7FF6C29E48A9
[7] menu\src\hooks\protections\received_clone_sync.cpp L: 27 big::hooks::received_clone_sync
[8]	GTA5.exe+0x16B9E11 0x7FF6C2F89E11
[9]	GTA5.exe+0x16C0476 0x7FF6C2F90476
[10]	GTA5.exe+0x13E973D 0x7FF6C2CB973D
[11]	GTA5.exe+0x16C05CD 0x7FF6C2F905CD
[12]	GTA5.exe+0x13E9B1A 0x7FF6C2CB9B1A
[13] menu\src\hooks\protections\receive_net_message.cpp L: 285 big::hooks::receive_net_message




[0] menu\src\services\players\player_service.cpp L: 57 big::player_service::get_by_host_token
[1] menu\src\hooks\info\get_network_event_data.cpp L: 88 big::hooks::get_network_event_data
[2]	GTA5.exe+0x1147BD2 0x7FF6C2A17BD2
[3]	GTA5.exe+0x113B05E 0x7FF6C2A0B05E
[4]	GTA5.exe+0x16AFC21 0x7FF6C2F7FC21
[5]	GTA5.exe+0x10DA101 0x7FF6C29AA101
[6]	GTA5.exe+0x11BEEE7 0x7FF6C2A8EEE7
[7]	GTA5.exe+0x16BA1F5 0x7FF6C2F8A1F5
[8]	GTA5.exe+0x11BB4E1 0x7FF6C2A8B4E1
[9]	GTA5.exe+0x16C2EC6 0x7FF6C2F92EC6
[10] menu\src\hooks\protections\received_event.cpp L: 698 big::hooks::received_event
[11]	GTA5.exe+0x16C38D4 0x7FF6C2F938D4
[12]	GTA5.exe+0x16BAC7D 0x7FF6C2F8AC7D
[13]	GTA5.exe+0x16C0325 0x7FF6C2F90325
[14]	GTA5.exe+0x13E973D 0x7FF6C2CB973D
[15]	GTA5.exe+0x16C05CD 0x7FF6C2F905CD
[16]	GTA5.exe+0x13E9B1A 0x7FF6C2CB9B1A
[17] menu\src\hooks\protections\receive_net_message.cpp L: 285 big::hooks::receive_net_message
```

**Comment by lonelybud** [comment]
today i got crashed again. I started new public session then few players got added and i got crashed with logs similar to above. I can see that the function call `player->get_net_data()` crashes the game. Since this time i am using custom flavor of menu i am therefore adding checks for null. Cant figure out why it gets null

**Comment by lonelybud** [comment]
sometime it looks like crash when console is in edit mode or ones leave the game paused maybe. I would like to close this for now. Cant reproduce anymore.
